### ISSUE TITLE
Spellcheck text selection styling doesn't look right in CupertinoTextField

### ISSUE THREAD

**Issue by leighajarett** [flutter/flutter#124404]
When you select a misspelled word in a CupertinoTextField, the styling is the same as other text selection (uses the little bars with the circle, and a color supplied by the theme). 

![Apr-07-2023 12-25-07](https://user-images.githubusercontent.com/21065911/230649847-f37fc737-c87b-40af-9944-299658b4b737.gif)

However, on a UIKit/SwiftUI app, if you select a misspelled word, no bars are used and the highlight color is red. 

![Hfgbgb](https://user-images.githubusercontent.com/21065911/230649909-9e65d928-e9ec-4486-8e39-a0a3abf246e8.jpeg)

When speaking with developers, text input is something that has been raised as important to "feel native." So, setting the default styling to match native iOS would be great. But, there might be cases where someone wants to customize this styling. Ideally we expose APIs in this widget to control the styling of the selection, and have the default values match those on iOS (maybe this already exists, but I couldn't tell how to do it from a quick glance). 

However, given that it's[ difficult to adap](https://github.com/flutter/flutter/issues/122647)t between CupertinoTextField and TextField, we might consider leveraging TextField for all platforms, but ensure we can add capabilities to match platform-specific styling and behaviors.

cc: @MitchellGoodwin @camsim99

**Comment by Renzo-Olivares** [comment]
cc @justinmc

**Comment by justinmc** [comment]
Thanks for this, you're right about the red highlight.

Is that red highlight actual selection, or is it something different?  It wouldn't be too hard for us to change the selection color to red and to hide the handles when tapping a misspelled word.  Is that what native iOS is doing though?

**Comment by gnprice** [comment]
At a quick test (in the Notes app, on iOS 15.6.1):
* If I double-tap on a misspelled word (meaning a word that has a red dotted underline), I get a text selection on it with the same styling as on any other word.
* But if I *single*-tap on a misspelled word, then I get a UI similar to the screenshot in the issue report:
  * a red-tinted highlight
  * no bars/handles
  * a menu that has the same style as a selection menu, but instead of the usual text selection menu's options it has a list of spelling replacements, or "No Replacements Found".

  (For me that menu has a black background and no shadow, but the same is true of a text selection menu. I assume the difference between that vs. the white background and shadow in the issue report is either an iOS version difference, or something in my device's settings.)

In short I think this is a feature request for providing iOS's spellcheck UI.

It looks like there's an existing issue which covers this, though it's kind of an umbrella issue:
* #34688

… Oh aha, and there's a more specific issue for this particular feature:
* #119537 

So this issue may be a duplicate of that one. And @camsim99 recently closed that issue in #119189, merged just a few days before this was filed.

@leighajarett Does this issue still reproduce for you on the main/master channel?

**Comment by justinmc** [comment]
@gnprice Thanks for trying this!  What happens if, after single tapping to show the red highlight, you type a letter on the keyboard?  I'm wondering if the red highlight will behave like selection or not. (I could try this myself later if you don't have time, I just have no Apple device on me at the moment.)

On master the experience should be better, but there is still no red highlight.

**Comment by gnprice** [comment]
> What happens if, after single tapping to show the red highlight, you type a letter on the keyboard?

Just tried that, and the result is that the new letter replaces the whole word. So in that respect it behaves like selection.

**Comment by justinmc** [comment]
Thanks!  That's actually easier to do I think.
